packages/transaction-controller/src/TransactionController.ts

```diff
@@ -556,14 +607,113 @@ export class TransactionController extends BaseControllerV1<
 
     this.addPendingTransactionTrackerListeners();
 
+    this.subscribe(this.#onStateChange);
+
+    this.messagingSystem.subscribe(
+      'NetworkController:stateChange',
+      (_, patches) => {
+        const networkClients = this.getNetworkClientRegistry();
+        patches.forEach(({ op, path }) => {
+          if (op === 'remove' && path[0] === 'networkConfigurations') {
+            const networkClientId = path[1] as NetworkClientId;
+            delete networkClients[networkClientId];
+          }
+        });
+
+        this.#refreshTrackingMap(networkClients);
+        this.#refreshEtherscanRemoteTransactionSources(networkClients);
+      },
+    );
+
     onNetworkStateChange(() => {
       log('Detected network change', this.getChainId());
+      // TODO(JL): Network state changes also trigger PendingTransactionTracker's onStateChange.
+      // Verify if this is still necessary when the feature branch is being reviewed
+      this.#onStateChange();
       this.onBootCleanup();
     });
 
     this.onBootCleanup();
+    this.#initTrackingMap();
   }
 
+  /**
+   * Stops polling and removes listeners to prepare the controller for garbage collection.
+   */
+  destroy() {
+    this.#stopAllTracking();
+  }
+
+  // TODO(JL): I think NetworkController should expose a NetworkClientRegistry type
+  #refreshEtherscanRemoteTransactionSources = (
+    networkClients: ReturnType<NetworkController['getNetworkClientRegistry']>,
+  ) => {
+    // this will be prettier when we have consolidated network clients with a single chainId:
+    // check if there are still other network clients using the same chainId
+    // if not remove the etherscanRemoteTransaction source from the map
+    const chainIdsInRegistry = new Set();
+    Object.values(networkClients).forEach((networkClient) =>
+      chainIdsInRegistry.add(networkClient.configuration.chainId),
+    );
+    const existingChainIds = Array.from(
+      this.etherscanRemoteTransactionSourcesMap.keys(),
+    );
+    const chainIdsToRemove = existingChainIds.filter(
+      (chainId) => !chainIdsInRegistry.has(chainId),
+    );
+
+    chainIdsToRemove.forEach((chainId) => {
+      this.etherscanRemoteTransactionSourcesMap.delete(chainId);
+    });
+  };
+
+  // TODO(JL): I think NetworkController should expose a NetworkClientRegistry type
+  #refreshTrackingMap = (
+    networkClients: ReturnType<NetworkController['getNetworkClientRegistry']>,
+  ) => {
```

Comment:
should we address this?

Reply:
Lets do it separately. Not required for this work

Reply:
Tracking here https://github.com/MetaMask/MetaMask-planning/issues/1019 and will remove comment

Reply:
https://github.com/MetaMask/core/pull/3643/commits/352575ac7cd3362b7bb97a9817395ce13ace90d3